src/neurogen/mesh.py

```diff
@@ -0,0 +1,566 @@
+def density_decomposition(vertices,
+                        faces,
+                        segment_id,
+                        directory,
+                        minimum_vertices=1024,
+                        quantization_bits=16,
+                        compression_level=5,
+                        mesh_subdirectory='meshdir'):
+
+    """ Generates a Neuroglancer precomputed multiresolution mesh based
+    on the number of vertices within each fragment
+    
+    Parameters
+    ----------
+    vertices : numpy array
+        Vertices to convert to trimesh object.
+    faces : numpy array
+        Faces to convert to trimesh object.
+    segment_id : str
+        The ID of the segment to which the mesh belongs. 
+    directory : str
+        Neuroglancer precomputed volume directory.
+    minimum : int
+        The minimum number of vertices that a fragment needs to have in order a
+        stop breaking into more octrees.  Default is 1024
+    quantization_bits : int
+        Number of bits for mesh vertex quantization. Can only be 10 or 16. 
+    compression_level : int
+        Level of compression for Draco format.
+    mesh_subdirectory : str
+        Name of the mesh subdirectory within the Neuroglancer volume directory.    
+    """
+
+    mesh = trimesh.Trimesh(vertices=vertices, faces=faces)
+    density_decomposition_mesh(mesh=mesh,
+                               segment_id=segment_id,
+                               minimum_vertices=minimum_vertices,
+                               quantization_bits=quantization_bits,
+                               compression_level=compression_level,
+                               mesh_subdirectory=mesh_subdirectory)
+
+def generate_mesh_dataframe(vertices, minvertices, lod=0):
+
+    """ This function generates a dataframe that contains information 
+    on the levels of details based on density.  The function bin the vertices 
+    to fit in the number nodes in the respective LOD. It will continue until
+    all fragments of the highest level of detail contains less than minvertices.
+    
+    Parameters
+    ----------
+    vertices : numpy array 
+        A 3x1 array of the mesh's vertices
+    minvertices : int
+        The maximum number of vertices within a fragment in 
+        the highest level of detail
+    lod : int
+        The current number of levels to create the mesh
+
+    Returns
+    -------
+    lod : int
+        The number of levels created for the mesh
+    mesh_lods : pandas dataframe
+        The dataframe containing information on the level of 
+        details
+    """
+    # Intialize necessary information
+    reachlevel = False 
+    num_vertices = len(vertices)
+
+    # Initalize necessary information to prevent recalculating
+    maxvertex = vertices.max(axis=0)
+    minvertex = vertices.min(axis=0)
+
+    # Initialize the dataframes
+    # mesh_df contains the orginial vertices of the mesh
+    mesh_df = pd.DataFrame(data=vertices, columns=["x", "y", "z"])
```

Comment:
Can we please do this without the pandas dependency?

Reply:
Do you have any recommendations as to how I should go about it?